app/services/SearchService.scala

```diff
@@ -2,17 +2,19 @@ package services
 
 import com.decodified.scalassh._
 import devsearch.ast.ContentsSource
+import play.api.cache.Cache
 import scala.io.Source
 import devsearch.parsers._
 import devsearch.features._
 import scala.util._
 import scala.io.Source
+import play.api.Play.current
 
 case class SearchResults(entries: Seq[SearchResultEntry])
 case class SearchResultEntry(repo: String, path: String, line: Int)
 
 object SearchService {
-  def get(query: String): SearchResults = {
+  def get(query: String): SearchResults = Cache.getOrElse(query, 3600){
```

Comment:
I still feel like this cache might kick us in the ass later when someone is not aware of it or has to turn it off manually. Also, the chances that we make the same query twice is fairly low, I think? Have you looked into what this could mean in terms of memory usage?